lib/webauthn/authenticator_attestation_response.rb

```diff
@@ -90,6 +94,25 @@ def valid_attestation_statement?
       @attestation_type, @attestation_trust_path = attestation_statement.valid?(authenticator_data, client_data.hash)
     end
 
+    def valid_attestation_trustworthiness?
+      case @attestation_type
+      when WebAuthn::AttestationStatement::ATTESTATION_TYPE_NONE
+        WebAuthn.configuration.acceptable_attestation_types.include?(:None)
+      when WebAuthn::AttestationStatement::ATTESTATION_TYPE_SELF
+        WebAuthn.configuration.acceptable_attestation_types.include?(:Self)
+      else
+        return false unless WebAuthn.configuration.acceptable_attestation_types.include?(:Basic)
+
+        trust_store = OpenSSL::X509::Store.new.tap do |store|
+          attestation_root_certificates_store.each do |cert|
+            store.add_cert(cert)
+          end
+        end
+
+        trust_store.verify(leaf_certificate, signing_certificates)
```

Comment:
Following on from my comment above, I'd recommend to use a StoreContext object here for verification as done in https://github.com/cedarcode/webauthn-ruby/pull/283/commits/6f7ac72cf53c2befea6ffb9cdfc48ec537ef831b#diff-1ddcb17dfc94f5bcdda73ce349e7b06e - maybe that actually should be the return value `valid_attestation_trustworthiness?`.

This way it is safe to have our resolver block use Store objects from constants in multi-threaded contexts, which would be preferable over reading from disk which can perform unpredictably when the system is under load.

Reply:
Putting aside the confusion that rises from the name, the idea behind the new configuration option is to set an object that implements a `find` method that returns an array with the trusted root certificates. Each time we perform trust path validation we call this method, get those certificates and create a `Store` which we use to verify the chain that one time. We create different trust stores for different responses. 

Do you see anything that compromises thread safety? I understand the use of `StoreContext` when the trust store being used is always the same, but here we create a different one for each request. I do believe that will be a concern once we offer default root certificates (like #283). 
Do you think the `find` method should return a trust store instead of the certificates?

Maybe there's some multithreading problem I'm missing, or the `attestation_root_certificates_store` naming confusion made the whole thing confusing, but either way the design is open to discussion so any insight you can provide is completely valuable. Maybe the idea will be clearer once I modify the [name of the methods](https://github.com/cedarcode/webauthn-ruby/pull/282#discussion_r354380752).